frontend/app/(modals)/comments.tsx

```diff
@@ -1,82 +1,376 @@
 import CommentComponent from '@/components/Comment';
-import { Container, Main } from '@/tamagui.config';
-import { Feather } from '@expo/vector-icons';
+import { UserContext } from '@/contexts/UserContext';
+import { CommentSchema, CommentSchemaInputs } from '@/types/commentInput';
+import {
+  AntDesign,
+  Feather,
+  MaterialCommunityIcons,
+  Octicons,
+} from '@expo/vector-icons';
+import { zodResolver } from '@hookform/resolvers/zod';
+import axios from 'axios';
 import { useLocalSearchParams, useRouter } from 'expo-router';
-import { Dimensions } from 'react-native';
-import { Button, ScrollView, Text, TextArea, XStack } from 'tamagui';
+import { useContext, useEffect, useState } from 'react';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import {
+  Dimensions,
+  KeyboardAvoidingView,
+  Modal,
+  useColorScheme,
+} from 'react-native';
+import GestureRecognizer from 'react-native-swipe-gestures';
+import { Button, Form, Input, ScrollView, Text, View } from 'tamagui';
 
 export default function CommentPage() {
-  const { name } = useLocalSearchParams();
+  //get user information
+  const { token, user_id } = useContext(UserContext);
+
+  //get parameters from the post
+  const item = useLocalSearchParams();
+  const stringedItem = JSON.stringify(item);
+  const params = JSON.parse(stringedItem.toString());
+
+  //router to move between modal and friends page
   const router = useRouter();
-  {
-    /*This is the back button that goes back to the home page*/
-  }
+
+  //states
+  const [comments, setComments] = useState([]);
+  const [goFetch, setGoFetch] = useState(true);
+  const [input, setInput] = useState('');
+  const [showDelete, setShowDelete] = useState(false);
+  const [enableShowDelete, setEnableShowDelete] = useState(false);
+
+  //refetch data
+  const refetch = async () => {
+    //try and make get call for the post of the comments
+    try {
+      const postCheck = await axios.get(
+        `${process.env.EXPO_PUBLIC_IP_ADDR}/api/${params['post_id'].slice(1, -1)}`,
+        { headers: { Authorization: `Bearer ${token}` } },
+      );
+      //return the comments of the post
+      return postCheck.data['comments'];
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  //if goFetch is true, refetch data
+  useEffect(() => {
```

Comment:
use useQuery to get the data, it will do it when you load the page so you don't need the goFetch boolean.